lib/sync/config.go

```diff
@@ -0,0 +1,80 @@
+package sync
+
+import (
+	"time"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/node"
+	"boscoin.io/sebak/lib/storage"
+	"github.com/inconshreveable/log15"
+)
+
+const (
+	SyncPoolSize             = 300
+	FetchTimeout             = 1 * time.Minute
+	RetryInterval            = 10 * time.Second
+	CheckBlockHeightInterval = 30 * time.Second
+)
+
+type Config struct {
+	storage           *storage.LevelDBBackend
+	network           network.Network
+	connectionManager network.ConnectionManager
+	networkID         []byte
+	localNode         *node.LocalNode
+	logger            log15.Logger
+	commonCfg         common.Config
+
+	SyncPoolSize             int
+	FetchTimeout             time.Duration
+	RetryInterval            time.Duration
+	CheckBlockHeightInterval time.Duration
+	Logger                   log15.Logger
+}
+
+func NewConfig(networkID []byte,
+	localNode *node.LocalNode,
+	st *storage.LevelDBBackend,
+	nt network.Network,
+	cm network.ConnectionManager,
+	cfg common.Config) *Config {
+	c := &Config{
+		storage:           st,
+		network:           nt,
+		connectionManager: cm,
+		logger:            log.New(log15.Ctx{"node": localNode.Alias()}),
+		networkID:         networkID,
+		commonCfg:         cfg,
+		localNode:         localNode,
+
+		SyncPoolSize:             SyncPoolSize,
+		FetchTimeout:             FetchTimeout,
+		RetryInterval:            RetryInterval,
+		CheckBlockHeightInterval: CheckBlockHeightInterval,
+	}
+	return c
+}
+
+func (c *Config) NewSyncer() *Syncer {
```

Comment:
Yeah I would not instantiate the Syncer from a config object, but make the function instantiating the syncer accept a config

Reply:
I dont like config as value for contracting. But it is my taste. And you are too. 
IMHO,From client view,Config.NewSyncer probably makes sence. 
Ps) Original name was Builder.

Reply:
ping @Geod24